plasmapy/simulation/particletracker.py

```diff
@@ -75,43 +85,57 @@ def __init__(self, plasma, particle_type='p', n_particles=1, scaling=1,
 
         self.plasma = plasma
 
-        self.dt = dt
+        self._dt = dt.si.value
         self.NT = int(nt)
-        self.t = np.arange(nt) * dt
+        self._t = np.arange(nt) * self._dt
 
-        self.x = np.zeros((n_particles, 3), dtype=float) * u.m
-        self.v = np.zeros((n_particles, 3), dtype=float) * (u.m / u.s)
+        self._x = np.zeros((n_particles, 3), dtype=float)
+        self._v = np.zeros((n_particles, 3), dtype=float)
         self.name = particle_type
 
-        self.position_history = np.zeros((self.NT, *self.x.shape),
-                                         dtype=float) * u.m
-        self.velocity_history = np.zeros((self.NT, *self.v.shape),
-                                         dtype=float) * (u.m / u.s)
-        # create intermediate array of dimension (nx,ny,nz,3) in order to allow
-        # interpolation on non-equal spatial domain dimensions
-        _B = np.moveaxis(self.plasma.magnetic_field.si.value, 0, -1)
-        _E = np.moveaxis(self.plasma.electric_field.si.value, 0, -1)
-
-        self._B_interpolator = interp.RegularGridInterpolator(
-            (self.plasma.x.si.value,
-             self.plasma.y.si.value,
-             self.plasma.z.si.value),
-            _B,
-            method="linear",
-            bounds_error=True)
-
-        self._E_interpolator = interp.RegularGridInterpolator(
-            (self.plasma.x.si.value,
-             self.plasma.y.si.value,
-             self.plasma.z.si.value),
-            _E,
-            method="linear",
-            bounds_error=True)
-
-    def _interpolate_fields(self):
-        interpolated_b = self._B_interpolator(self.x.si.value) * u.T
-        interpolated_e = self._E_interpolator(self.x.si.value) * u.V / u.m
-        return interpolated_b, interpolated_e
+        self._position_history = np.zeros((self.NT, *self.x.shape),
+                                          dtype=float)
+        self._velocity_history = np.zeros((self.NT, *self.v.shape),
+                                          dtype=float)
+        self._hqmdt = (self.eff_q / self.eff_m / 2 * dt).si.value
```

Comment:
I am wondering if there is a better variable name than `hqmdt`.

Reply:
There most certainly is. But if I wrote "half charge timestep over mass", line length would explode in a few places.

On the other hand, I have been reading Computer Simulation Using Particles recently, so my thinking may be contaminated by the FORTRAN examples...